subprojects/model-core/src/main/java/org/gradle/api/internal/IssueReporter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal;
+
+import org.gradle.internal.service.scopes.EventScope;
+import org.gradle.internal.service.scopes.Scopes;
+
+/**
+ * Collects issues to be reported to users.
+ */
+@EventScope(Scopes.Build.class)
+public interface IssueReporter {
```

Comment:
Design idea: I considered a new purposed listener interface (like `UnsafeCredentialsListener`) but felt this (reporting issues that potentially break configuration cache) is such a common pattern that could warrant a general protocol and avoid us having to create a new interface every time we need that.

Reply:
Two considerations: 
* The proposed pattern doesn't provide enough information for the compiler to warn us when a new introduced issue is not handled at the listener site (like adding a new abstract method to an existing interface does, for example);
* I wouldn't introduce a different pattern now just for this one issue;
The search for a simpler mechanism here has merit but IMO we should approach it holistically and intently.